client/store/auth.js

```diff
@@ -1,61 +1,90 @@
-import axios from 'axios'
-import history from '../history'
+import axios from 'axios';
+import history from '../history';
 
-const TOKEN = 'token'
+const TOKEN = 'token';
 
 /**
  * ACTION TYPES
  */
-const SET_AUTH = 'SET_AUTH'
+const SET_AUTH = 'SET_AUTH';
+const LOGOUT = 'LOGOUT';
 
 /**
  * ACTION CREATORS
  */
-const setAuth = auth => ({type: SET_AUTH, auth})
+const setAuth = (auth) => ({ type: SET_AUTH, auth });
 
 /**
  * THUNK CREATORS
  */
-export const me = () => async dispatch => {
-  const token = window.localStorage.getItem(TOKEN)
+export const me = () => async (dispatch) => {
+  const token = window.localStorage.getItem(TOKEN);
   if (token) {
     const res = await axios.get('/auth/me', {
       headers: {
-        authorization: token
-      }
-    })
-    return dispatch(setAuth(res.data))
+        authorization: token,
+      },
+    });
+    return dispatch(setAuth(res.data));
   }
-}
+};
 
-export const authenticate = (email, password, method) => async dispatch => {
+export const authenticate = (email, password, method) => async (dispatch) => {
   try {
-    const res = await axios.post(`/auth/${method}`, {email, password})
-    window.localStorage.setItem(TOKEN, res.data.token)
-    dispatch(me())
+    const res = await axios.post(`/auth/${method}`, { email, password });
+    window.localStorage.setItem(TOKEN, res.data.token);
+    dispatch(me());
     history.push('/home');
   } catch (authError) {
-    return dispatch(setAuth({error: authError}))
+    return dispatch(setAuth({ error: authError }));
   }
-}
+};
+
+export const login = (password, email, method) => async (dispatch) => {
+  try {
+    const res = await axios.post(`/auth/${method}`, { password, email });
+    window.localStorage.setItem(TOKEN, res.data.token);
+    dispatch(me());
+  } catch (authError) {
+    return dispatch(setAuth({ error: authError }));
+  }
+};
+
+export const signup =
+  (email, password, firstName, lastName, method) => async (dispatch) => {
```

Comment:
same here the method is always going to be `signup` if you're routing it to the signup thunk creator